compiler/rustc_const_eval/src/errors.rs

```diff
@@ -87,3 +87,27 @@ pub(crate) struct TransientMutBorrowErrRaw {
     pub span: Span,
     pub kind: ConstContext,
 }
+
+#[derive(SessionDiagnostic)]
+#[error(const_eval::const_evaL_max_num_nodes_exceeded)]
+pub(crate) struct MaxNumNodesExceeded {
+    #[primary_span]
+    pub span: Span,
+    pub s: String,
+}
+
+#[derive(SessionDiagnostic)]
+#[error(const_eval::unallowed_fn_pointer_call)]
+pub(crate) struct UnallowedFnPointerCall {
+    #[primary_span]
+    pub span: Span,
+    pub kind: ConstContext,
+}
+
+#[derive(SessionDiagnostic)]
+#[error(const_eval::unstable_const_fn)]
+pub(crate) struct UnstableConstFn {
+    #[primary_span]
+    pub span: Span,
+    pub def_id: String,
```

Comment:
This should at least be called `def_path`

Reply:
We really should work out a way to just keep a `DefId` in these types and have `IntoDiagnosticArg` call `def_path_str` for us, like we discussed at RustConf, @compiler-errors.

Reply:
> This should at least be called `def_path`

Renaming it at the next commit